### PR TITLE
Slower damaged units for Dune 2k mod

### LINKED ISSUE #19454: D2K Add smoke effects on health lower than 50%
REQUEST MOTIVATION

In the original Dune 2000 when vehicles are damaged they start to move slower. This is noticeable in the campaign missions because it makes it harder to rush into the enemy territory bypassing their defenses.

PROPOSED SOLUTION

The system should work as follows:
* When vehicle health is lower than 50%, it starts emitting smoke particles.
* The smoke effect ends after some time
* The effect should resume in case of any negative health update (excludes repairing).
* When below 50% health, movement speed is reduced (regardless of whether the smoke animation is playing)

### PR DESCRIPTION
**Pull request by Leo512bit**
A simple PR that will partly resolve #19454 however the smoke particles not implanted in this PR just units slowing when damaged.

### CODE REVIEW

File: mods/d2k/rules/defaults.yaml [hunk 1]
```diff
@@ -241,6 +241,12 @@
 			Radius: 16
 	MapEditorData:
 		Categories: Vehicle
+	GrantConditionOnDamageState@HEAVY:
+		Condition: heavy-damage
+		ValidDamageStates: Heavy
```
Comment (abcdefg30) [inline mods/d2k/rules/defaults.yaml]: I think we should remove this line. The defaults are `Heavy` and `Critical` already. Defined like this heavily damaged units are slowed down, but units in the red are not!

**Review by abcdefg30 — COMMENTED** [review]
Lgtm otherwise. We can do the smoke in a different PR.

**Review by Leo512bit — COMMENTED** [review]
Fixed the mistake I did with `GrantConditionOnDamageState`

**Comment by abcdefg30** [comment]
Can you please squash the commits?

**Comment by Leo512bit** [comment]
I'll just make a new PR for some reason I can't squash it

**Pull request closed without merging**